game-app/game-core/src/main/java/games/strategy/triplea/odds/calculator/ConcurrentBattleCalculator.java

```diff
@@ -79,12 +83,15 @@ public void setGameData(final GameData data) {
         latchSetData.countDown();
       } else {
         cancelCurrentOperation.incrementAndGet();
-        // increment our token, so that we can set the data in a different thread and return from
-        // this one
-        latchWorkerThreadsCreation.increment();
-        AsyncRunner.runAsync(() -> createWorkers(data))
-            .exceptionally(
-                throwable -> log.error("Error when trying to create Workers", throwable));
+        // assign the future,
+        // so that we can set the data in a different thread wait for its completion
+        latchWorkerThreadsCreation =
+            CompletableFuture.runAsync(() -> createWorkers(data))
+                .exceptionally(
+                    throwable -> {
+                      log.error("Error when trying to create Workers", throwable);
+                      return null;
+                    });
```

Comment:
The exceptionally block is pretty verbose with CompletableFuture. WDYT of adding an overload that doesn't require a return value so it can be as concise as the one with AsyncRunner.runAsync()?

Reply:
To be honest I almost thing this should be made synchronous. If the callers support it this would simplify the code quite a lot